### PR TITLE
WEB-271 migrate to dataAttributes

### PR DESCRIPTION
**Pull request by pladaria**
instead of defining `data-whatever` props in our components, we want to use a dictionary `dataProperties` to be able to pass any `data attribute` to the component

### CODE REVIEW

File: src/touchable.tsx [hunk 1]
```diff
@@ -126,16 +127,6 @@ export interface PropsMaybeOnPress extends CommonProps {
 
 type Props = PropsHref | PropsTo | PropsOnPress | PropsMaybeHref | PropsMaybeTo | PropsMaybeOnPress;
 
-const getPrefixedDataAttributes = (attrs?: DataAttributes) => {
```
Comment (pladaria) [inline src/touchable.tsx:129]: moved to utils/dom.tsx

File: src/box.tsx [hunk 2]
```diff
@@ -23,8 +26,11 @@ type Props = {
     children?: React.ReactNode;
     className?: string;
     role?: string;
+    // @deprecated use dataAttributes
     'data-testid'?: string;
+    // @deprecated use dataAttributes
     'data-qsysid'?: string;
+    dataAttributes?: DataAttributes;
```
Comment (atabel) [inline src/box.tsx:33]: did you miss the jsdoc here?

**Review by atabel — APPROVED** [review]

**Review by anabelengp — APPROVED** [review]

File: src/box.tsx (hunk 2 above)
Comment (pladaria) [reply src/box.tsx:33]: added

**Pull request merged**